crates/sparrow-runtime/src/execute.rs

```diff
@@ -118,10 +118,18 @@ impl ExecutionOptions {
         self.changed_since_time = changed_since;
     }
 
+    pub fn set_changed_since_s(&mut self, seconds: i64) {
+        self.changed_since_time = Timestamp { seconds, nanos: 0 };
+    }
+
     pub fn set_final_at_time(&mut self, final_at_time: Timestamp) {
         self.final_at_time = Some(final_at_time);
     }
 
+    pub fn set_final_at_time_s(&mut self, seconds: i64) {
+        self.final_at_time = Some(Timestamp { seconds, nanos: 0 });
+    }
+
```

Comment:
nit: now that we have multiple methods that set this with different time units, I'd think it better practice to change these two parameters to private, and add access methods. i.e. I wouldn't want to be able to bypass these methods and directly mutate the variable, in case I set it using the wrong time unit somehow. *Admittedly low risk since you have to convert to `Timestamp` before anyway, so not a big deal.

Reply:
Maybe? I look at this as a class that should probably go away once we have the Python Session stuff cleaned up, so don't know that we want to maintain the old execution path that much.